AEPMessaging/Sources/Messaging+PublicAPI.swift

```diff
@@ -91,4 +94,76 @@ import UserNotifications
+    /// If the feeds content for one or more surface paths isn't previously cached in the SDK, it will not be retrieved from the Adobe Journey Optimizer via the Experience Edge network.
+    /// - Parameters:
+    ///   - surfacePaths: An array of surface path strings.
+    ///   - completion: The completion handler to be invoked with a dictionary containing the surface paths and the corresponding Feed objects.
+    static func getFeedsForSurfacePaths(_ surfacePaths: [String], _ completion: @escaping ([String: Feed]?, Error?) -> Void) {
+        let validSurfacePaths = surfacePaths
+            .filter { !$0.isEmpty }
+
+        guard !validSurfacePaths.isEmpty else {
+            Log.warning(label: MessagingConstants.LOG_TAG,
+                        "Cannot get feeds as the provided surface paths array has no valid items.")
+            completion(nil, AEPError.invalidRequest)
+            return
+        }
+
+        let eventData: [String: Any] = [
+            MessagingConstants.Event.Data.Key.GET_FEEDS: true,
+            MessagingConstants.Event.Data.Key.SURFACES: validSurfacePaths
+        ]
+
+        let event = Event(name: MessagingConstants.Event.Name.GET_MESSAGE_FEEDS,
+                          type: EventType.messaging,
+                          source: EventSource.requestContent,
+                          data: eventData)
+
+        MobileCore.dispatch(event: event) { responseEvent in
+            guard let responseEvent = responseEvent else {
+                completion(nil, AEPError.callbackTimeout)
+                return
+            }
+
+            if let error = responseEvent.responseError {
+                completion(nil, error)
+                return
+            }
+
+            guard let feeds = responseEvent.feeds else {
+                completion(nil, AEPError.unexpected)
+                return
+            }
+
+            completion(feeds, .none)
+        }
+    }
+
+    /// Registers a permanent event listener with the Mobile Core for listening to personalization decisions events received upon a personalization query to the Experience Edge network.
+    /// - Parameter completion: The completion handler to be invoked with a dictionary containing the surface paths and the corresponding Feed objects.
+    static func setFeedsHandler(_ completion: (([String: Feed]) -> Void)? = nil) {
+        if !isFeedResponseListenerRegistered {
+            isFeedResponseListenerRegistered = true
+            MobileCore.registerEventListener(type: EventType.messaging, source: EventSource.notification, listener: feedsResponseListener(_:))
+        }
+        feedsResponseHandler = completion
+    }
+
+    private static func feedsResponseListener(_ event: Event) {
+        guard let feedsResponseHandler = feedsResponseHandler else {
+            return
+        }
+
+        guard
+            let feeds = event.feeds,
+            !feeds.isEmpty
```

Comment:
The difference in the behavior is because it doesn't make sense to invoke streaming listener with empty response but when explicitly requested using get API then we need to return whatever's available or error. Empty response in the case of get API might also imply we do not have any feeds cached for the requested surface as opposed to an error occurred. There are many ways to handle this e.g. we can update AEPError for some additional scenarios say 'contentUnavailable' or define extension specific Error enum and then return appropriate response.